src/test/scala/com/microsoft/hyperspace/index/IndexManagerTests.scala

```diff
@@ -399,13 +401,46 @@ class IndexManagerTests extends HyperspaceSuite with SQLHelper {
           case Seq(
               OptimizeActionEvent(_, _, "Operation started."),
               OptimizeActionEvent(_, _, msg)) =>
-            assert(msg.contains("Optimize aborted as no index files smaller than 1 found."))
+            assert(msg.contains("Optimize aborted as no applicable files smaller than 1 found."))
           case _ => fail()
         }
       }
     }
   }
 
+  test("Verify optimize is a no-op if it is not needed.") {
```

Comment:
Can you make the test name a bit more clear? sth like "optimize is no-op if each bucket has a single index file.".